### ISSUE TITLE
Creating image views should be less restrictive if VK_IMAGE_CREATE_EXTENDED_USAGE_BIT is used

### ISSUE THREAD

**Issue by pdagobert** [KhronosGroup/Vulkan-ValidationLayers#1495]
I have a compute shader which writes compressed pixels into a compressed texture.  It works, but it triggers the following error:

`vkCreateImageView(): pCreateInfo->format  with tiling VK_IMAGE_TILING_OPTIMAL does not support usage that includes VK_IMAGE_USAGE_STORAGE_BIT. The Vulkan spec states: If usage contains VK_IMAGE_USAGE_STORAGE_BIT, then the image view's format features must contain VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT. (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-VkImageViewCreateInfo-usage-02275)`

This is made possible by two flags introduced in [vk_khr_maintenance2](https://github.com/KhronosGroup/Vulkan-Docs/blob/master/appendices/VK_KHR_maintenance2.txt).

VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT flag allows to create an image view with an uncompressed format of an image with a compressed format.

VK_IMAGE_CREATE_EXTENDED_USAGE_BIT allows to create images with incompatible format/usages. This is necessary because compressed formats doesn't allow the storage bit. Which means it would be impossible to create the image, but it would also be impossible to create an image view with the same format as the compressed image, used only for reading.

This problem is referenced by #96. Assuming the above example is correct according to the specs, there is no way for the validation layers to detect a misused flag without false positives. VK_IMAGE_CREATE_EXTENDED_USAGE_BIT should only be used in order to lift this kind of restrictions, so shouldn't validation layers do the same?

**Comment by mark-lunarg** [comment]
```
 VUID-VkImageCreateInfo-flags-01572
If flags contains VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT, then format must be a
block-compressed image format, an ETC compressed image format, or an ASTC
compressed image format.
• VUID-VkImageCreateInfo-flags-01573
If flags contains VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT, then flags must also
contain VK_IMAGE_CREATE_MUTABLE_FORMAT_BIT.
• VUID-VkImageViewCreateInfo-image-01583
If image was created with the VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT flag,
format must be compatible with, or must be an uncompressed format that is sizecompatible with, the format used to create image.
• VUID-VkImageViewCreateInfo-image-01584
If image was created with the VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT flag, the
levelCount and layerCount members of subresourceRange must both be 1.
```

**Comment by danginsburg** [comment]
I'm hitting this issue too, using Validation layers v1.2.162.0.  Same exact use case as the original bug reporter.

**Comment by mark-lunarg** [comment]
@pdagobert, @danginsburg, will bump in priority.

**Comment by mark-lunarg** [comment]
@danginsburg, this is now at the top of my list -- I would very much appreciate the gfxrecon trace, please.

**Comment by danginsburg** [comment]
@mark-lunarg Sorry, this slipped through the cracks on me this week.  I'll get you the trace on Monday, please ping me by e-mail if I forget.

**Comment by danginsburg** [comment]
@mark-lunarg Pinging on this issue - still seeing it with Vulkan SDK 1.2.176.1.  I believe I privately sent you a trace that showed the issue?  If not, I can send one again.

**Comment by ncesario-lunarg** [comment]
@danginsburg  I'm traveling today, but I will hook up with Mark this week to get the trace.

**Comment by mark-lunarg** [comment]
Dan, forwarded the trace to Nathaniel.

**Comment by ncesario-lunarg** [comment]
Thanks @mark-lunarg!

**Comment by ncesario-lunarg** [comment]
Sorry, I bumped this one towards the back of the wagon so many times it eventually _fell off_ the wagon.

I'm not sure the original VUID (02275) can be reproduced, at least not without hitting `02251` first. I've put up a change (#3662) that addresses the CTS error, and I _think_ it should resolve the one @danginsburg is seeing as well, but I'm having hardware issues at the moment trying to get the original replay to work. I'm not 100% sure my change is correct, though; I think it is, but the worry is that by using `VK_IMAGE_CREATE_EXTENDED_USAGE_BIT` one could create an image view later on that fails in the driver without triggering a validation message. I need to do some more testing on my part to try and verify that is working correctly.

**Comment by ncesario-lunarg** [comment]
@aitor-lunarg is also looking into this.

**Comment by sfricke-samsung** [comment]
So I was able to run `dEQP-VK.image.sample_texture.128_bit_compressed_format` and get the `VUID-VkImageCreateInfo-imageCreateMaxMipLevels-02251` error because it doesn't support the `VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT` bit

> VK_IMAGE_CREATE_EXTENDED_USAGE_BIT allows to create images with incompatible format/usages. This is necessary because compressed formats doesn't allow the storage bit. Which means it would be impossible to create the image, but it would also be impossible to create an image view with the same format as the compressed image, used only for reading.

I guess I don't see this as a Validation Layer issue... `VK_IMAGE_CREATE_EXTENDED_USAGE_BIT` was added in `VK_KHR_maintenance2` for

> Allow creating images with usage flags that may not be supported for the base image’s format, but are supported for image views of the image that have a different but compatible format.

so if there is not compatible format for the compressed format with the usage flag, then the test will need to be changed as the VVL are doing things correct (in this case) IMO

**Comment by sfricke-samsung** [comment]
So I pushed my attempt to fix this by just ignoring the the `VkImage` Creation Limits when using `VK_IMAGE_CREATE_EXTENDED_USAGE_BIT` (same idea as https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3662)

I was able to pass

```
./deqp-vk --deqp-case=dEQP-VK.image.sample_texture.*_bit_compressed_format*
```

now with no validation errors

As for issues with errors in at `vkCreateImageView` time, I still take the stance that there is no bug as you need to be using a supported `ImageView::format` regardless of the `VK_IMAGE_CREATE_*` flags used

**Comment by aitor-lunarg** [comment]
Regarding the initial purpose of the issue, there may be 2 reasons to why that error is triggered. Having in mind that usage is inherited from the image:

1. The usage is not compatible with the image view's format. Even if the image was created with the extended usage and block texel view compatible, when we create the view, the view's usage and format need to be compatible. A compressed format won't allow for storage usage. We would require a format that is storage compatible for that usage.

2. This happens with the provided trace by @danginsburg. Having a compressed format image with extended usage and block texel view compatible, if we want to create a view to sample that same compressed format, we need to restrict the usage of the view because compressed formats do not allow for storage. This is done with `VkImageViewUsageCreateInfo` passed in `VkImageViewCreateInfo`'s `pNext`. If we don't have this, we will get the error from the layer since the usage (inherited from the image) and format are not compatible.

I have added a test to showcase how to correctly create the views if there's any doubt

**Comment by ncesario-lunarg** [comment]
Thanks for the confirmation @danginsburg. It does seem like `VK_KHR_image_format_list` and `VK_IMAGE_CREATE_EXTENDED_USAGE_BIT_KHR` aren't fully compatible, would you agree @sfricke-samsung?

**Comment by ncesario-lunarg** [comment]
Closing this for now and creating a new issue to capture what is fixed in #3724 (per Aitor's suggestion).
